crypto/pkcs11key/key.go

```diff
@@ -252,10 +312,117 @@ func getPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkc
 	if !gotModulus || !gotExponent {
 		return noKey, errors.New("public key missing either modulus or exponent")
 	}
-	return rsa.PublicKey{
+
+	rsa := rsa.PublicKey{
 		N: n,
 		E: e,
-	}, nil
+	}
+	return rsa, nil
+}
+
+// Get the public key matching an Elliptic Curve private key
+func (ps *Key) getECPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkcs11.ObjectHandle) (interface{}, error) {
+	var noKey interface{}
+
+	template := []*pkcs11.Attribute{
+		pkcs11.NewAttribute(pkcs11.CKA_EC_PARAMS, nil),
+		pkcs11.NewAttribute(pkcs11.CKA_ID, nil),
+	}
+
+	attr, err := module.GetAttributeValue(session, privateKeyHandle, template)
+	if err != nil {
+		return noKey, err
+	}
+
+	oid := []byte {}
+	id := []byte {}
+	gotOid, gotID := false, false
+	for _, a := range attr {
+		if a.Type == pkcs11.CKA_EC_PARAMS {
```

Comment:
Also, is CKA_EC_PARAMS generally present on both the private and public key, or just the private one?